harness/determined/cli/dev.py

```diff
@@ -1,17 +1,24 @@
 import argparse
+import collections
+import inspect
+import json
+import re
 import shlex
 import shutil
 import subprocess
 import sys
+import typing
```

Comment:
this was mainly here to let me `eval` the annotations. I moved it down there. I do like sticking to `typing` and not importing the type hints directly but much more if we alias it to something short same as `numpy as np` otherwise a compound type gets way too long and we need more type aliases and so on.
I updated it